test/jdk/java/security/KeyStore/CheckMacOSKeyChainIntermediateCATrust.java

```diff
@@ -0,0 +1,204 @@
+                "-k", loginKeyChain,
+                caPath.toString()
+        );
+        executeProcess(args);
+
+        caPath = Path.of("%s/%s".formatted(DIR, "non-trusted-test-ca.pem"));
+        args = List.of(
+                "/usr/bin/security",
+                "add-certificates",
+                "-k", loginKeyChain,
+                caPath.toString()
+        );
+        executeProcess(args);
+
+        caPath = Path.of("%s/%s".formatted(DIR, "test-intermediate-ca.pem"));
+        args = List.of(
+                "/usr/bin/security",
+                "add-certificates",
+                "-k", loginKeyChain,
+                caPath.toString()
+        );
+        executeProcess(args);
+
+    }
+
+    private static String getLoginKeyChain() {
+        return Path.of(System.getProperty("user.home"), "Library/Keychains/login.keychain-db").toString();
+    }
+
+    private static void executeProcess(List<String> params) {
+        System.out.println("Command line: " + params);
+        try {
+            ProcessTools.executeProcess(params.toArray(new String[0]))
+                    .shouldHaveExitValue(0);
+        } catch (Exception e) {
+            fail("Unexpected exception: " + e);
+        }
+    }
+
+    private static void deleteCertificatesFromKeyChain() {
+        executeProcess(
+                List.of(
+                        "/usr/bin/security",
+                        "delete-certificate",
+                        "-c", "Non Trusted Example CA",
+                        "-t"
+                )
+        );
+
+        executeProcess(
+                List.of(
+                        "/usr/bin/security",
+                        "delete-certificate",
+                        "-c", "Example CA",
+                        "-t"
+                )
+        );
+
+        executeProcess(
+                List.of(
+                        "/usr/bin/security",
+                        "delete-certificate",
+                        "-c", "Example Intermediate CA",
+                        "-t"
+                )
+        );
+    }
+
+    private static boolean not(boolean condition) {
+        return !condition;
+    }
+
+    private static boolean containsSubjectName(List<X509Certificate> certificates, String subjectName) {
+        return certificates.stream()
+                .map(cert -> cert.getSubjectX500Principal().getName())
+                .anyMatch(name -> name.contains(subjectName));
+    }
+
+    private static List<String> getSubjects(List<X509Certificate> certificates) {
+        return certificates.stream()
+                .map(cert -> cert.getSubjectX500Principal().getName())
+                .toList();
+    }
+
+    private static void assertThat(boolean expected, String message, List<X509Certificate> certificates) {
+        if (!expected) {
+            String errMessage = message;
+            if (verbose) {
+                errMessage += ", subjects: " + getSubjects(certificates);
```

Comment:
By default, subjects and elements in the subjects are separated by a comma. Can you change the separator between subjects?